components/eventing-controller/api/v1alpha2/errors.go

```diff
@@ -18,6 +18,7 @@ var (
 	NSPath     = field.NewPath("metadata").Child("namespace")
 
 	EmptyErrDetail          = "must not be empty"
+	InvalidURIErrDetail     = "must be a valid URI reference"
```

Comment:
Are we asking user to provide URI reference? We are mostly expecting just a string as source? Maybe it can cause confusion for the user?

Reply:
a URI reference is also a string with certain limitations. The source should be a valid URI reference. That's how the error will look like.

Reply:
Agreed to string: `must be a valid source as per RFC 3986`